types/src/transaction/mod.rs

```diff
@@ -297,13 +309,24 @@ impl RawTransaction {
 
 #[derive(Clone, Debug, Hash, Eq, PartialEq, Serialize, Deserialize)]
 pub enum TransactionPayload {
-    WriteSet(ChangeSet),
+    WriteSet(WriteSetPayload),
     /// A transaction that executes code.
     Script(Script),
     /// A transaction that publishes code.
     Module(Module),
 }
 
+#[derive(Clone, Debug, Hash, Eq, PartialEq, Serialize, Deserialize)]
+pub enum WriteSetPayload {
+    /// Directly passing in the write set.
+    Direct(ChangeSet),
+    /// Generate the writeset by running a script.
+    WriteSetScript {
+        signer: AccountAddress,
```

Comment:
Nits: 
- Perhaps rename to something like `execute_as` or `to_imitate` to convey that this isn't an ordinary use of `signer`?
- Could you amend the doc comment for `WriteSetScript` to explain what execution does with this address?